### ISSUE TITLE
[TEST] Add unit/integration/end-to-end tests

### ISSUE THREAD

**Issue by starbops** [starbops/kubevirtbmc#6]
Currently, the project lacks testing to protect the code. We need to add different levels of testing to ensure the code's robustness and quality. This is also a requirement before we can transfer the project to the KubeVirt organization according to the [implementation phases of the design proposal](https://github.com/kubevirt/community/blob/main/design-proposals/kubevirtbmc.md#phase-1-transfer-the-kubebmc-project-to-the-kubevirt-organization).

P.S. This is a follow-up issue of #1.

**Comment by starbops** [comment]
I didn't notice that the linked PR #8 closed the issue automatically. I've added more tests in #29 and am still working on the e2e tests.

**Comment by starbops** [comment]
We still lack the e2e testing for the data plane, i.e., virtual machine interaction via IPMI (and Redfish, hopefully).
